### PR TITLE
executor: use pdqsort to improve performance

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/34250

Problem Summary:

### What is changed and how it works?

- use pdqsort to sort  https://github.com/golang/go/issues/50154, In common patterns, pdqsort is often faster (i.e. 10x faster in sorted slices).
- remove ```github.com/cznic/sortutil```

### Check List

Tests 

- [x] Unit test
- [x] Integration test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/hash_table.go [hunk 1]
```diff
@@ -86,9 +86,6 @@ type hashRowContainer struct {
 
 	rowContainer *chunk.RowContainer
 	memTracker   *memory.Tracker
-
```
Comment (wshwsh12) [inline executor/hash_table.go:89]: Why need modify this file?

Comment (hawkingrei) [reply executor/hash_table.go:89]: It may be copied from the old branch. I will fix it.

Comment (hawkingrei) [reply executor/hash_table.go]: Done

**Comment by cfzjywxk** [comment]
Is there any benchmark test information for TiDB after this subsitution?

**Pull request closed without merging**